analyzer/src/funTest/kotlin/CargoSubcrateTest.kt

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.analyzer
+
+import com.here.ort.analyzer.managers.Cargo
+import com.here.ort.downloader.VersionControlSystem
+import com.here.ort.model.yamlMapper
+import com.here.ort.utils.normalizeVcsUrl
+import com.here.ort.utils.test.DEFAULT_ANALYZER_CONFIGURATION
+import com.here.ort.utils.test.DEFAULT_REPOSITORY_CONFIGURATION
+import com.here.ort.utils.test.USER_DIR
+import com.here.ort.utils.test.patchExpectedResult
+
+import io.kotlintest.matchers.beEmpty
+import io.kotlintest.should
+import io.kotlintest.shouldBe
+import io.kotlintest.shouldNotBe
+import io.kotlintest.specs.StringSpec
+
+import java.io.File
+
+class CargoSubcrateTest : StringSpec() {
+    private val projectDir = File(
+        "src/funTest/assets/projects/synthetic/cargo-subcrate"
+    ).absoluteFile
+    private val vcsDir = VersionControlSystem.forDirectory(projectDir)!!
+    private val vcsUrl = vcsDir.getRemoteUrl()
+    private val vcsRevision = vcsDir.getRevision()
+
+    init {
+        "Lib project dependencies are detected correctly" {
+            val packageFile = File(projectDir, "Cargo.toml")
+            val vcsPath = vcsDir.getPathToRoot(projectDir)
+            val expectedResult = patchExpectedResult(
+                File(projectDir.parentFile, "cargo-subcrate-lib-expected-output.yml"),
+                definitionFilePath = "$vcsPath/Cargo.toml",
+                path = vcsPath,
+                revision = vcsRevision,
+                url = normalizeVcsUrl(vcsUrl)
+            )
+
+            val result = createCargo().resolveDependencies(listOf(packageFile))[packageFile]
+
+            result shouldNotBe null
+            result!!.errors should beEmpty()
+            yamlMapper.writeValueAsString(result) shouldBe expectedResult
+        }
+    }
+
+    init {
```

Comment:
I wasn't aware it even works to have multiple `init` blocks 😄 But for consistency we should merge these into only a single `init` block.